### PR TITLE
Add Jenkins plugin to Showcase

### LINKED ISSUE #173: Add Jenkins plugin
### Acceptance criteria

We need to add the [Jenkins plugin](https://github.com/backstage/backstage/tree/master/plugins/jenkins). 

The Overview card (EntityLatestJenkinsRunCard) should be added to the 'Overview' tab in the Entity page and wrapped with the conditional `isJenkinsAvailable ` flag.

The Jenkins card (EntityJenkinsContent) should be added to the 'CI/CD' tab in the Entity page and wrapped with the conditional `isJenkinsAvailable ` flag.

### PR DESCRIPTION
**Pull request by RobotSail**
Signed-off-by: Oleg S <97077423+RobotSail@users.noreply.github.com>

## Description

This pull request adds the Jenkins plugin to 
the backstage showcase code. The user can 
configure their Jenkins instance via 
`app-config.yaml` by providing their:

- `JENKINS_URL`
- `JENKINS_USERNAME`
- `JENKINS_API_TOKEN`

The Jenkins plugin adds a history of runs to 
the CI/CD page and an overview of the jenkins 
entity to the overview page.

#### Overview tab:

image

#### CI/CD tab:
image

## Which issue(s) does this PR fix

Fixes #173

## PR acceptance criteria

Please make sure that the following steps are complete:

- GitHub Actions are completed and successful
- Unit Tests are updated and passing
- E2E Tests are updated and passing
- Documentation is updated if necessary
- Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -67,6 +67,11 @@ proxy:
     allowedMethods: ['GET']
     auth: ${SONARQUBE_TOKEN}
 
+  '/jenkins/api':
+    target: ${JENKINS_URL}
+    headers:
+      Authorization: ${JENKINS_API_TOKEN}
```
Comment (schultzp2020) [inline app-config.yaml:73]:
```suggestion
      Authorization: ${JENKINS_TOKEN}
```
Rename to be consistent with other env vars

File: app-config.yaml [hunk 2]
```diff
@@ -161,6 +166,13 @@ sonarqube:
   baseUrl: ${SONARQUBE_URL}
   apiKey: ${SONARQUBE_TOKEN}
 
+jenkins:
+  instances:
+  - name: default
+    baseUrl: ${JENKINS_URL}
+    username: ${JENKINS_USERNAME}
+    apiKey: ${JENKINS_API_TOKEN}
```
Comment (schultzp2020) [inline app-config.yaml:174]:
```suggestion
    apiKey: ${JENKINS_TOKEN}
```

File: packages/app/src/App.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { Route } from 'react-router-dom';
+
```
Comment (schultzp2020) [inline packages/app/src/App.tsx:3]:
```suggestion
```
Remove extra line

File: packages/app/src/App.tsx [hunk 4]
```diff
@@ -34,12 +31,18 @@ import { CatalogGraphPage } from '@backstage/plugin-catalog-graph';
 import { RequirePermission } from '@backstage/plugin-permission-react';
 import { catalogEntityCreatePermission } from '@backstage/plugin-catalog-common/alpha';
 import { CssBaseline, ThemeProvider } from '@material-ui/core';
-import { janusTheme } from './themes/janus';
-import { HomePage } from './components/home/HomePage';
 import { HomepageCompositionRoot } from '@backstage/plugin-home';
 import { OcmPage } from '@janus-idp/backstage-plugin-ocm';
+
```
Comment (schultzp2020) [inline packages/app/src/App.tsx:36]:
```suggestion
```

File: packages/app/src/App.tsx [hunk 5]
```diff
@@ -34,12 +31,18 @@ import { CatalogGraphPage } from '@backstage/plugin-catalog-graph';
...
+import { apis } from './apis';
 import Logo from './components/Root/LogoIcon';
+
```
Comment (schultzp2020) [inline packages/app/src/App.tsx:39]:
```suggestion
```

File: packages/app/src/components/catalog/EntityPage/Content/Overview.tsx [hunk 6]
```diff
@@ -182,5 +186,11 @@ export const overviewContent = (
         </Grid>
       </EntitySwitch.Case>
     </EntitySwitch>
+
+    <EntitySwitch>
+      <EntitySwitch.Case if={isJenkinsAvailable}>
+        <EntityLatestJenkinsRunCard branch="" />
```
Comment (schultzp2020) [inline packages/app/src/components/catalog/EntityPage/Content/Overview.tsx:192]:
```suggestion
        <Grid item xs={12}>
          <EntityLatestJenkinsRunCard branch="" />
        </Grid>
```
Wrap the Jenkins card with a grid item

Comment (schultzp2020) [inline packages/app/src/components/catalog/EntityPage/Content/Overview.tsx:192]: Does the `branch` prop need to be specified?

**Review by schultzp2020 — CHANGES_REQUESTED** [review]
Don't forget to update `showcase-docs/getting-started.md` as well.

File: packages/app/src/components/catalog/EntityPage/Content/Overview.tsx (hunk 6 above)
Comment (RobotSail) [reply packages/app/src/components/catalog/EntityPage/Content/Overview.tsx:192]: Yeah, it's a required prop. I left it blank to specify all branches.

**Comment by RobotSail** [comment]
@schultzp2020 I made the requested changes. The only one that I can't change is the `branch=""` prop on the `< EntityLatestJenkinsRunCard />` card since it's required.

**Comment by schultzp2020** [comment]
@RobotSail if that is the case lets go with `branch="main,master"` as the default.

**Comment by schultzp2020** [comment]
@RobotSail Don't forget to update `showcase-docs/getting-started.md` as well.

**Comment by RobotSail** [comment]
Ah yes, not sure how I missed that. Updated now @schultzp2020

**Review by schultzp2020 — DISMISSED** [review]
LGTM

**Comment by RobotSail** [comment]
@schultzp2020 I had to rebase and it seems like it invalidated your review

**Review by schultzp2020 — APPROVED** [review]
LGTM

**Pull request merged**